File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        prefix = [0] 
        n = len(nums)
        for i in range(n):
            prefix.append(prefix[-1] + nums[i])

        count = 0
        for i in range(1, n):
            if (2 * prefix[i] - prefix[n]) % 2 == 0:
                count += 1
        return count
        

File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int answer = 0; 
        int left =0;
        int total = 0;
        for (int k = 0; k < nums.length; k ++){
            total += nums[k];
        }
        if (total % 2 != 0){
            return answer;
        }
        for (int i =0; i < nums.length - 1 ; i++ ){
             left += nums[i];
             int right = total - left; 
            
        int difference = right - left ;
        if (difference % 2 == 0){
            answer += 1;
        }
        }
        return answer;
    }
}